diff -Naur /home/pearu/src/omniscidb-5.7.0-orig/Embedded/DBEngine.cpp /home/pearu/src/omniscidb-5.7.0/Embedded/DBEngine.cpp
--- /home/pearu/src/omniscidb-5.7.0-orig/Embedded/DBEngine.cpp	2021-08-27 23:00:24.000000000 +0300
+++ /home/pearu/src/omniscidb-5.7.0/Embedded/DBEngine.cpp	2021-09-06 20:46:39.074320003 +0300
@@ -148,7 +148,7 @@
 
     try {
       db_handler_ =
-          mapd::make_shared<DBHandler>(prog_config_opts.db_leaves,
+          std::make_shared<DBHandler>(prog_config_opts.db_leaves,
                                        prog_config_opts.string_leaves,
                                        prog_config_opts.base_path,
                                        prog_config_opts.allow_multifrag,
@@ -426,7 +426,7 @@
  private:
   std::string base_path_;
   std::string session_id_;
-  mapd::shared_ptr<DBHandler> db_handler_;
+  std::shared_ptr<DBHandler> db_handler_;
   bool is_temp_db_;
   std::string udf_filename_;
 
